pkg/utils/secrets/certificate.go

```diff
@@ -62,7 +62,11 @@ type CertificateSecretConfig struct {
 	SigningCA *Certificate
 	PKCS      int
 
-	Validity                          *time.Duration
+	Validity *time.Duration
+	// RenewAfterValidityPercentage sets the percentage of the validity when the certificate should be renewed.
+	// The effective check for renewal is after the given percentage of validity or 10d before the end of validity.
+	// If not specified the default percentage is 80.
+	RenewAfterValidityPercentage      *int
```

Comment:
I think this is the wrong place to add this configuration. We should rather move it to the `GenerateOptions` https://github.com/gardener/gardener/blob/41a694b024144465bd4f02f39d6d1c3493549c2f/pkg/utils/secrets/manager/generate.go#L342-L360 just like the `Validity` field. This way, the feature can also be used for other secret types, not only certificates.

Please also define a function like this: https://github.com/gardener/gardener/blob/41a694b024144465bd4f02f39d6d1c3493549c2f/pkg/utils/secrets/manager/generate.go#L500-L507

and perhaps augment the unit tests to also other secret types.